specs/_features/eip7547/fork-choice.md

```diff
@@ -0,0 +1,77 @@
+# EIP-7547 -- Fork Choice
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Helpers](#helpers)
+  - [New `verify_inclusion_list`](#new-verify_inclusion_list)
+  - [New `is_inclusion_list_available`](#new-is_inclusion_list_available)
+- [New fork-choice handlers](#new-fork-choice-handlers)
+  - [New `on_inclusion_list`](#new-on_inclusion_list)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice accompanying EIP-7547.
+
+## Helpers
+
+### New `verify_inclusion_list`
+
+```python
+def verify_inclusion_list(state: BeaconState, block: BeaconBlock, inclusion_list: InclusionList,
+                          execution_engine: ExecutionEngine) -> bool:
+    """
+    returns true if the inclusion list is valid. 
+    """
+    # Check that the inclusion list corresponds to the block proposer
+    signed_summary = inclusion_list.signed_summary
+    proposer_index = signed_summary.message.proposer_index
+    assert block.proposer_index == proposer_index
+
+    # Check that the signature is correct
+    assert verify_inclusion_list_summary_signature(state, signed_summary)
```

Comment:
We are calling `verify_inclusion_list_summary_signature` twice. I'm not sure if it's necessary.

Reply:
removed! ty